The Update

skyboy opened the pull request.

simon816 commented:
Is this ever going to get pulled or should @skyboy's fork be the only place for active development?

vmrob commented:
@simon816 If @skyboy wanted to he could include a LICENSE file with the source code and include the terms of the license extracted from the [official forums](http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/1287424-1-5-1-1-5-2-powercrystals-mods-the-updates-never#licence) authored by @powercrystals:

> License (all mods)
> Free for non-commercial use. You are free to include any of my mods in any modpacks or server-specific packs without needing to ask permission (but it'd be nice to hear if you do!), provided something either in the download or on your site lists me as the creator. You are also free to produce Bukkit ports. A link to this thread would be nice for any of the above, but that's all I ask.
> 
> You also have implied permission to build mods that depend on PowerCrystalsCore, if you want some of the functionality in there.

I would probably modify it slightly for inclusion on GitHub and include relevant credits in a README.

> Copyright 2014 PowerCrystals and contributors
> 
> Free for non-commercial use. You are free to include this mod in any modpacks or server-specific packs provided the above copyright notice is maintained. You are also free to produce Bukkit ports.
> 
> You also have implied permission to build mods that depend on PowerCrystalsCore.

With that said, if this repository is no longer active, I think the absolute best course of action would be to contact @powercrystals and see if there's an open source license such as the MIT license that he/she would be willing to apply to the project. I know I'm not a contributor at all, but I do have interest in bettering the open source community. I'll see what I can do to get in contact with @PowerCrystals.

powercrystals commented:
Though I'm not exactly active anymore I still get notifications from this tracker (and then immediately ignore them), so congratulations on successfully if perhaps inadvertently contacting me. For your actual question, the MIT license would be fine by me. I am not going to do any further commits one way or the other though so as to avoid giving people hope that I'm returning, so hopefully this comment is sufficient because there will never be a LICENSE or README in any of these repositories. If not, uh, oh well?

vmrob commented:
@powercrystals Thanks for that comment! I know the community really appreciates your contributions and I'm personally really glad to see that you're providing permission to apply the MIT license to your work. I'll migrate this conversation to the appropriate repo (skyboy/MineFactoryReloaded).

I went ahead and made a [pull request](https://github.com/powercrystals/MineFactoryReloaded/pull/382) for the addition of README/LICENSE files for the sake of clarity and to (hopefully) redirect future visitors the appropriate repo. If you change your mind about adding those, I wanted to make it as effortless as possible for you.

Thanks again!

skyboy committed:
Fix grinders setting a world to easy when peaceful

skyboy committed:
Reduce WorldGenMassiveTree memory allocation

P3pp3rF1y committed:
Chorus support in picker and planter

P3pp3rF1y committed:
removed prePick and postPick in these

skyboy committed:
Update buildfile's CoFH maven and add curse upload

skyboy committed:
Merge pull request #589 from P3pp3rF1y/1.10.x

1.10 Update

skyboy committed:
Apply curseforge plugin to buildscript

skyboy committed:
Apply it somewhere else

skyboy committed:
Wiki of curseforge plugin is unclear on where things are to go

skyboy committed:
Those arguments were wrong

skyboy committed:
Given up on getting gradle to work on jenkins

skyboy committed:
Fun with case sensitivity

skyboy committed:
Update BON, exclude forestry/ic2 integration

skyboy committed:
Comment IItemDuct logic

skyboy committed:
Label the beta as a beta

skyboy committed:
Copy dependencies to where reobf can see them

skyboy committed:
Update bon to handle lambdas

skyboy committed:
Package only the correct classes

Really need to get other mod's APIs out of the repo

skyboy committed:
Change gradle script to what it was previously

skyboy committed:
Remove long-defunct mfr-forestry.revision

P3pp3rF1y committed:
Updated dependency versions

P3pp3rF1y committed:
Added item handler to DSU so that mods that use item handler can work with it (AE2)

P3pp3rF1y committed:
Using item.placeBlockAt instead of simply setting blockstate as block placement can fail in some cases. Fixes #576

P3pp3rF1y committed:
cleaned unused imports

skyboy committed:
Convert bucket overwrite to RegistryUtils

P3pp3rF1y committed:
Merge branch 'master' into 1.10.x

skyboy committed:
Merge pull request #602 from P3pp3rF1y/1.10.x

Dependency updates, DSU item handler and block placer dupe bug fix (still happened in 1.10)

skyboy committed:
Move AT and manifest back to META-INF

skyboy committed:
Fluids properly vaproize through the provided APIs

skyboy committed:
Clean up imports

skyboy committed:
Fountains respects fluids vaproizing via provided APIs

skyboy committed:
Clean up unifier logic a little

skyboy committed:
Remove custom language parsing logic in favor of forge's

skyboy committed:
Rename language package to lang for normal loading

P3pp3rF1y committed:
Fixed EnderIO recipes

skyboy committed:
Merge pull request #626 from P3pp3rF1y/1.10.x

Fixed EnderIO recipes

P3pp3rF1y committed:
updated core version

P3pp3rF1y committed:
Fixed two word fluid names that now have underscore in them

P3pp3rF1y committed:
ItemHandler cap support for machines and conveyor

skyboy committed:
Merge pull request #629 from P3pp3rF1y/1.10.x

cofhcore version update and fix for fluid names

P3pp3rF1y committed:
some changes

P3pp3rF1y committed:
more progress

wilkemeyer committed:
Added check to relightChunk that verifies that the given chunk has skylight data stored

P3pp3rF1y committed:
it builds and runs now!

skyboy committed:
Merge pull request #632 from wilkemeyer/bugfix-worldgencrash-noskylightdim

Added check to relightChunk that verifies that the given chunk has skylight data stored

skyboy committed:
Resolve #611

P3pp3rF1y committed:
All the null stuff for ItemStacks

P3pp3rF1y committed:
Merge remote-tracking branch 'mfr/master' into 1.11-port

P3pp3rF1y committed:
And Nonnulllist for machine inventory and more null stuff for ItemStacks that was somehow hidden

P3pp3rF1y committed:
Changes related to update to latest CCL

P3pp3rF1y committed:
Fluid handler update for items and a fix for machine bucket handling

P3pp3rF1y committed:
and fixed the implementation to actually use ITankContainerBucketable because I am derp

P3pp3rF1y committed:
fixed child mod modids

P3pp3rF1y committed:
fixed stuffed items handling

P3pp3rF1y committed:
Minor fix to stop NPEs when fisher loads NBT during world load

P3pp3rF1y committed:
Some conveyor fixes

P3pp3rF1y committed:
Fixed tanks not connecting to plastic pipes when replaced

P3pp3rF1y committed:
Moved ITickable so that only tickable TEs get it and migrated over to onLoad instead of firstick cofh_validate

P3pp3rF1y committed:
Fixed steam boiler consuming water even when full of steam

P3pp3rF1y committed:
Added Items.BUCKET as container item so that this is returned in recipes that use milk bucket. Fixes #641

P3pp3rF1y committed:
Merge remote-tracking branch 'origin/1.10.x' into 1.11-port

skyboy committed:
Merge pull request #642 from P3pp3rF1y/1.10.x

Added Items.BUCKET as container item

P3pp3rF1y committed:
Fixed auto brewer

P3pp3rF1y committed:
Fixed fountain to only vaporize fluids in dimensions where water vaporizes

P3pp3rF1y committed:
fixed tree gen and decaying logic

skyboy committed:
Fix auto-brewer not functioning with forge's 1.10 logic

P3pp3rF1y committed:
auto brewer fixes to consume one ingredient per 2 potions

skyboy committed:
Clean and fix Auto-Brewer's logic

skyboy committed:
Fix fisher pausing at random

P3pp3rF1y committed:
passenger dropoff refactoring to prefer to search for suitable places from the rail out

P3pp3rF1y committed:
Removing unncecessary AT for canTriggerWalking and related change to make it protected in entityneedle

P3pp3rF1y committed:
Fixed RedstoneNode hashcode so that when node changes it can actually be found in Set.
Also fixed an issue where subnets retained power after a color was changed on the only input for the subnet.

P3pp3rF1y committed:
And made the implementation simpler because I am a derp

P3pp3rF1y committed:
"fixed" issue with rednet logic not taking input or providing output to different sides than were set in GUI

P3pp3rF1y committed:
Fixed saving/restoring powered state to/from meta

P3pp3rF1y committed:
merged 1.10 fixes

P3pp3rF1y committed:
merged 1.10 fixes

P3pp3rF1y committed:
Fixed cobblestone upgrade recipe

P3pp3rF1y committed:
Fixed crash with rednet meter and debug item when clicking on rednet cable

P3pp3rF1y committed:
Fixed display of entity name

P3pp3rF1y committed:
Fixed to show resourcelocation id of entity in "ID" row instead of the previous entity name

P3pp3rF1y committed:
Fixed always displaying name tag when entities are spawned in with fancy jailer's net.
(also no need to set name tag anymore as this is handled by entity deserialization code from NBT)

P3pp3rF1y committed:
Fixed porta spawner tooltip to display entities that spawner can spawn (and now can actually display multiple if the spawner is set to display more than one entity)

P3pp3rF1y committed:
Fixed porta spawner converting minecart to spawner minecart

P3pp3rF1y committed:
Changed xp extractor to point at player in hand

P3pp3rF1y committed:
Allow drinking from tank with straw

P3pp3rF1y committed:
Fixed picking up liquids and drinking those with factory cup

P3pp3rF1y committed:
Removed unused imports

P3pp3rF1y committed:
fixed spyglass raytrace and added display of state properties as well as preventing spam in chat

P3pp3rF1y committed:
Fixed potato canon to be able to find ammo in survival and made it not shoot the shooting player when hitting very close entity

P3pp3rF1y committed:
Fixed active status of machine to be shown with delay same as it did in 1.7.10

P3pp3rF1y committed:
Fixed plastic pipe causing a lot of client lag, because it kept updating the chunk for rendering every single tick

P3pp3rF1y committed:
minor update to have spyglass use getActualState for block state.

P3pp3rF1y committed:
fixed modid (lower case now)

P3pp3rF1y committed:
Commented out MFR milk bucket - forge replacement on the way

DYColdWind committed:
Update zh_CN.lang

skyboy committed:
Merge pull request #681 from DYColdWind/zh_CN.lang

Update zh_CN.lang

P3pp3rF1y committed:
Fixed rednet controller not taking/sending signal to sides entered in GUI. Fixes #685

P3pp3rF1y committed:
Fixed position that's used in cansustainplant check. Fixes crash reported in #680

P3pp3rF1y committed:
Fixed auto brewer not making water bottles and a bit of refactoring for better readability (merging stuff valid for 1.10 from 1.11). Plus keeping track of spare resources between world saves.

P3pp3rF1y committed:
Updated to newer versions of forge and cofhcore

P3pp3rF1y committed:
Fixed PRC manual. Fixes #672

P3pp3rF1y committed:
Fixed auto enchanter not keeping level setting between loads when set to 30. Fixes #666

P3pp3rF1y committed:
Added null world check. Fixes #665

P3pp3rF1y committed:
Fixed grinder not getting player only drops from mobs (like blaze rods). One of the issues mentioned in #659

P3pp3rF1y committed:
Fixed dupe bug in advanced collector recipe. Fixes #655

P3pp3rF1y committed:
Fixed modid checked in FactoryRegistry. Fixes #653

P3pp3rF1y committed:
Updated with changes discussed in code review

P3pp3rF1y committed:
Changed smashing world to use a simple IBlockAccess implementation, fixes block smasher crash #674

skyboy committed:
Merge pull request #690 from P3pp3rF1y/1.10.x

1.10 fixes

P3pp3rF1y committed:
1.12 port

P3pp3rF1y committed:
1.12 port, removing buildcraft api and properly referring to it instead

P3pp3rF1y committed:
1.12 port

P3pp3rF1y committed:
1.12 rendering changes

P3pp3rF1y committed:
World gen feature reload fix

P3pp3rF1y committed:
Fixed xp extractor not consuming buckets and a bit of related refactoring

P3pp3rF1y committed:
Fixed moving players on conveyors and tuned y calculation to be smoother

P3pp3rF1y committed:
Fixed item collector rendering

P3pp3rF1y committed:
Fixed some NPE crashes with containers and auto enchanter not working

P3pp3rF1y committed:
Fixed tank's break model and got rid of calling deprecated CCBakeryModel constructor.

P3pp3rF1y committed:
Fixed glitchy item in auto anvil's output slot that only existed on client

P3pp3rF1y committed:
Fixed issue with network sometimes not updating power level in case input node was removed

P3pp3rF1y committed:
Fixed cargo pickup rail to properly move items into chest minecart

P3pp3rF1y committed:
Enabled milk bucket

skyboy committed:
Testing with travis

skyboy committed:
Stop, it's compile time

skyboy committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into pr/700

# Conflicts:
#	gradle.properties
#	src/powercrystals/minefactoryreloaded/api/FactoryRegistry.java
#	src/powercrystals/minefactoryreloaded/block/ItemBlockRedNetLogic.java
#	src/powercrystals/minefactoryreloaded/core/GrindingDamage.java

#	src/powercrystals/minefactoryreloaded/gui/control/ButtonLogicBufferSelect.java
#	src/powercrystals/minefactoryreloaded/setup/recipe/Vanilla.java

#	src/powercrystals/minefactoryreloaded/tile/machine/TileEntityAutoBrewer.java

#	src/powercrystals/minefactoryreloaded/tile/machine/TileEntityBlockSmasher.java

#	src/powercrystals/minefactoryreloaded/tile/machine/TileEntityFisher.java

#	src/powercrystals/minefactoryreloaded/tile/machine/TileEntityGrinder.java
#	src/powercrystals/minefactoryreloaded/world/SmashingWorld.java

skyboy committed:
Merge branch 'refs/heads/pr/700'

skyboy committed:
Change these block comments

skyboy committed:
Fix blocks showing up twice in creative tabs

skyboy committed:
Fix items showing up in every creative tab

Also fix the hammer showing up in the delete slot. Is that a tab?

skyboy committed:
Fix plastic bags not displaying inventory contents

skyboy committed:
Refactor machines into sub-packages

skyboy committed:
SafariNets rename released mobs again

skyboy committed:
Single-use SafariNets stack to 12 again

skyboy committed:
Fix Historian being weird

skyboy committed:
Fix ejector not working on chunk load

skyboy committed:
Fix DSU not saving locked state

skyboy committed:
Rewrite SafariNet RandomMob logic

skyboy committed:
Fix returning the incorrect entity in some RandomMobProviders

skyboy committed:
Add new mobs to vanilla mob provider

skyboy committed:
Fix errors in some modhelpers

skyboy committed:
Update project files

skyboy committed:
Fix exact-spawning horses, make parrots ranchable

skyboy committed:
Fix machine recipies accepting PRC housings

skyboy committed:
Fix item/block mapping logic

skyboy committed:
Use registry events for item/block/model registration

skyboy committed:
Shift around villager trades

skyboy committed:
Shift properties out of MFR Core and into MFR Props

skyboy committed:
Don't crash on null tiles in the renderer

skyboy committed:
Fix TileEntity IDs and loading from older versions

skyboy committed:
Re-directorize everything

skyboy committed:
Try altering travis build

skyboy committed:
Try something else for travis

skyboy committed:
Things are more working on gradle front

skyboy committed:
Tweak zombie syringe

skyboy committed:
Properly package 'core' logic (oops)

skyboy committed:
Process fixes for MFR's inventories

skyboy committed:
Add DataFixer for SafariNet

skyboy committed:
Handle passengers for SafariNet correctly

skyboy committed:
Add PortaSpawner walker

skyboy committed:
Auto-formatting weirdness

skyboy committed:
Add remaning DataFixers

skyboy committed:
Get ASM core into functional state

skyboy committed:
Fix metadata for old SafariNets propagating past DataFixer

skyboy committed:
Sort transformer to really far in the list

skyboy committed:
Drop World requirement to release version

skyboy committed:
Require forge 14.23.4

skyboy committed:
Get source sets setup for compiling

skyboy committed:
Testing on travis

skyboy committed:
Modify .gitignore

skyboy committed:
Compile and package MFR-Core

skyboy committed:
Change build logic for contained dep

skyboy committed:
Get grinder/etc. completely functional

skyboy committed:
Fix an error in WorldTransformer

skyboy committed:
Fix WorldProxy proxing methods it ought not

skyboy committed:
Tweak GrindingWorldServer

skyboy committed:
Fix #702: ForgeGradle is broken

skyboy committed:
Fix threaded crash for grid updates

skyboy committed:
Bring steam boiler more in line with other fuel consumers

skyboy committed:
Fix worldgen errors

Closes #703

skyboy committed:
Add profiler debugging code

skyboy committed:
Remove `flipped` trees conceptually

The reworked harvest logic no longer cares, so we don't need to keep
this logic around.

skyboy committed:
Fix fence connectivity of RedNet cable

skyboy committed:
Fix random item registration order

skyboy committed:
Use `getItemBurnTime` instead of IFuelHandler

skyboy committed:
Use static event handler in places

skyboy committed:
Fix fuel handler

skyboy committed:
Change rocket lock-on logic

skyboy committed:
Initial redesign of the API

skyboy committed:
Clean up some logic.

skyboy committed:
Don't run RecipeSets' `clinit` until after preconditions are validated

skyboy committed:
Change localizations

skyboy committed:
Tweak recipe grouping

skyboy committed:
More localization

skyboy committed:
Prefix that. Oops.

skyboy committed:
Setup recipe groupings

skyboy committed:
Advancement files

skyboy committed:
Re-add restriction on upgrade crafting

skyboy committed:
Clean up TE integrator

skyboy committed:
Clean up readability of machine names in recipe class

skyboy committed:
Clean up formatting in recipe class

skyboy committed:
Tweak some vanilla recipes

skyboy committed:
Add back Thermal Expansion recipes

skyboy committed:
Update some things for compiling purposes

skyboy committed:
Fix bad localization key

skyboy committed:
Add package-info to API

skyboy committed:
Modify recipe grouping again

skyboy committed:
Add `@DependsOn` and `@After` to Integrators

skyboy committed:
Fix some javac warnings

skyboy committed:
Use LinkedHashList instead of LinkedHashSet

skyboy committed:
Use some Java8 features

skyboy committed:
Handle some code maturity inspections

skyboy committed:
Diamond operator

skyboy committed:
More maturity issues

skyboy committed:
More maturity stuff

skyboy committed:
Fix writing AttributeModifiers

skyboy committed:
Fix bug in Conveyors

skyboy committed:
Fix some errors in AutoDisenchanter

skyboy committed:
Fix an error in GrindableEnderman

skyboy committed:
Format GrindableEnderman

skyboy committed:
Remove superfluous null logic

skyboy committed:
Spelling pass

skyboy committed:
Split main and api sources, build better jars

skyboy committed:
Make Integrators able to throw

skyboy committed:
Remove defunct integration

skyboy committed:
Remove another defunct integrator

skyboy committed:
Transition integrators to new system

skyboy committed:
Change biome name lists to sets of biome ids

skyboy committed:
Also here

skyboy committed:
API formatting

skyboy committed:
Move FruitPicker's PlaySounds logic

skyboy committed:
Remove this annotation

skyboy committed:
Move MFRFarmables from `farmables` to `setup`

skyboy committed:
Extract ReplacementBlock to IReplacementBlock

skyboy committed:
Move compat modules into their own source set

skyboy committed:
Shuffle some classes around

skyboy committed:
Move base implementations into API

skyboy committed:
Remove compat dependencies on main

skyboy committed:
Annotate ILiquidDrinkHandler

skyboy committed:
Add base-line BC integration back

skyboy committed:
Simplify rotate logic in hammer

skyboy committed:
Remove redundant value check in RedNet cable

skyboy committed:
Fix ObjectHolder crashing the game by not using it

skyboy committed:
Replace `Map` with `IFactorySettings` in Harvestables

skyboy committed:
Fix compiling stuff

skyboy committed:
Update Harvester's API

skyboy committed:
Update Planter's API

skyboy committed:
Fix harvester saving and loading state

skyboy committed:
Remove `HarvestMode` from `TreeHarvestManager` as it was no longer utilized
